pysd/py_backend/functions.py

```diff
@@ -940,11 +1042,30 @@ def lookup_discrete(x, xs, ys):
     for index in range(0, len(xs)):
         if x < xs[index]:
             return ys[index - 1] if index > 0 else ys[index]
-    return ys[len(ys) - 1]
+    return ys[-1]
 
 
 def if_then_else(condition, val_if_true, val_if_false):
-    return np.where(condition, val_if_true, val_if_false)
+    """
+    Implements Vensim's IF THEN ELSE function.
+
+    Parameters
+    ----------
+    condition: bool or xarray.DataArray of bools
+    val_if_true: float/bool or xarray.DataArray
```

Comment:
Do we have tests in https://github.com/SDXorg/test-models that evaluate `if_then_else`, `xidz`, `zidz` with subscripts? Need to be careful that the operations are applied element-wise appropriately. Are there any cases where you might have a condition with fewer dimensions than the `val_if_true`, etc?

Reply:
in 
```
integration_test_vensim_pathway.py: test_subscripted_if_then_else
test-models/tests/subscripted_if_then_else/test_subscripted_if_then_else.mdl
integration_test_vensim_pathway.py: test_subscripted_xidz
test-models/tests/subscripted_xidz/test_subscripted_xidz.mdl
```
All the possible combinations of different dimensions are tested for if then else, zidz (in xidz test) and xidz

Reply:
Thanks!